src/melee.cpp

```diff
@@ -179,8 +210,39 @@ bool player::handle_melee_wear( item &shield, float wear_multiplier )
         g->m.add_item_or_charges( pos(), elem );
     }
 
+    // Preserve item temporarily for component breakdown
+    item temp = shield;
+
     remove_item( shield );
 
+    // Breakdown fragile weapons into components
+    if( temp.has_flag( "FRAGILE_MELEE" ) && !temp.components.empty() ) {
+        add_msg_player_or_npc( m_bad, _( "Your %s breaks apart!" ),
+                                      _( "<npcname>'s %s breaks apart!" ),
+                                      str.c_str() );
+
+        std::vector<item> all_comps = temp.components;
+
+        for( auto &comp : all_comps ) {
+            int break_chance = comp.typeId() == weak_comp ? 2 : 8;
+
+            if( one_in( break_chance ) ) {
+                add_msg_if_player( m_bad, _( "The %s is destroyed!" ), comp.tname() );
+                continue;
+            }
+
+            if( comp.has_flag( "HANDLE" ) && !is_armed() ) {
```

Comment:
Rather than explicit HANDLE flag, which would require a lot of marking, you could default to the largest item.
It would be correct in majority of cases and the minority could be special cased.